### PR TITLE
initialization, scaling, and pitt-peters solving tests

### PR DESCRIPTION
**Pull request by rcleuthold**
the same pull request as before...

### CODE REVIEW

**Comment by rcleuthold** [comment]
all tests run.

**Comment by jdeschut** [comment]
Hey hey! Thanks for all the additional clean-up work in this PR! Two small remarks/suggestions:

- why implement the `wound_tether_inequality` as an extra constraint, whereas one could directly set an upper bound to `l_t`?
- there are plenty of whitespace corrections in the diff of this PR, which makes it difficult to see at a glance what the real changes are, which makes the reviewing process time-consuming. Maybe in the future you can add the whitespace corrections to a separate feature request.  You can pick out changes you want to commit in a single file with `git add --patch mdl/dynamics.py` and then select those that you actually want to commit.

File: awebox/mdl/atmosphere.py [hunk 1]
```diff
@@ -47,9 +47,9 @@ def get_temperature(self, zz):
         elif options['model'] == 'windshear':
             t = params['t_ref'] - params['gamma_air'] * zz
         elif options['model'] == 'log_wind':
-            t = cas.DM(params['t_ref'])
+            t = params['t_ref'] * cas.DM.ones((1, 1))
```
[the remark this answers is not in the document]
Comment (rcleuthold) [reply awebox/mdl/atmosphere.py:50]: if I remember correctly: for some reason, the previous structure was giving me error messages... I didn't stay to debug - just found a working fix. If you have a better way, you're welcome to update it, too...

File: awebox/mdl/dynamics.py [hunk 2]
```diff
@@ -49,16 +49,18 @@
 import awebox.mdl.aero.tether_dir.coefficients as tether_drag_coeff
 
 import awebox.tools.vector_operations as vect_op
-
 import awebox.tools.struct_operations as struct_op
+import awebox.tools.print_operations as print_op
+
+import pdb
```
[the remark this answers is not in the document]
Comment (rcleuthold) [reply awebox/mdl/dynamics.py]: done!

**Comment by rcleuthold** [comment]
> * why implement the `wound_tether_inequality` as an extra constraint, whereas one could directly set an upper bound to `l_t`?

The idea was to give an alternative to the open-system lagrangian_momentum_correction concept, by closing the system: that is, specifically adding the mass of the amount-of-wound-up-tether to the mass of the generator drum. But, this means that you need to have some estimate of the maximum length that the tether ever reaches. My little solution is to make one more system parameter (in theta) that describes the full tether length value (called l_t_full), and then constrain that l_t(time) <= l_t_full... 

I'm not particularly attached to this solution - if you have a better idea? 

> * there are plenty of whitespace corrections in the diff of this PR, which makes it difficult to see at a glance what the real changes are, which makes the reviewing process time-consuming. Maybe in the future you can add the whitespace corrections to a separate feature request.  You can pick out changes you want to commit in a single file with `git add --patch mdl/dynamics.py` and then select those that you actually want to commit.

for the future, I will try my best!

File: test/reg/.cache/v/cache/lastfailed [hunk 3]
```diff
@@ -1,3 +1 @@
-{
```
[the remark this answers is not in the document]
Comment (rcleuthold) [reply test/reg/.cache/v/cache/lastfailed:1]: yes, sorry, this was an accident. I think I've removed the lastfailed files...

**Comment by rcleuthold** [comment]
so, the last change was to remove the lastfailed cache files. officially the tests are running again. but i have a very hard time imagining that they will fail... i'll post when they've run through.

**Comment by jdeschut** [comment]
> > * why implement the `wound_tether_inequality` as an extra constraint, whereas one could directly set an upper bound to `l_t`?
> 
> The idea was to give an alternative to the open-system lagrangian_momentum_correction concept, by closing the system: that is, specifically adding the mass of the amount-of-wound-up-tether to the mass of the generator drum. But, this means that you need to have some estimate of the maximum length that the tether ever reaches. My little solution is to make one more system parameter (in theta) that describes the full tether length value (called l_t_full), and then constrain that l_t(time) <= l_t_full...
> 
> I'm not particularly attached to this solution - if you have a better idea?

Hey, I overlooked that `l_t_full` was in `theta`, I thought I'd seen it in the fixed numerical parameter struct. All good!

**Comment by jdeschut** [comment]
Closing the system is definitely the more elegant solution, kudos to you!

**Comment by rcleuthold** [comment]
all tests passed. :)

**Pull request merged**